hack/.ci/lib/e2e.sh

```diff
@@ -0,0 +1,252 @@
+#!/bin/bash
+#
+# Copyright (C) 2024 ScyllaDB
+#
+
+set -euExo pipefail
+shopt -s inherit_errexit
+
+source "$( dirname "${BASH_SOURCE[0]}" )/../../lib/kube.sh"
+
+# gather-artifacts is a self sufficient function that collects artifacts without depending on any external objects.
+function gather-artifacts {
+  kubectl create namespace gather-artifacts --dry-run=client -o=yaml | kubectl_apply -f=-
+  kubectl create clusterrolebinding gather-artifacts --clusterrole=cluster-admin --serviceaccount=gather-artifacts:default --dry-run=client -o=yaml | kubectl_apply -f=-
+  kubectl create -n=gather-artifacts pdb must-gather --selector='app=must-gather' --max-unavailable=0 --dry-run=client -o=yaml | kubectl_apply -f=-
+
+  kubectl_create -n=gather-artifacts -f=- <<EOF
```

Comment:
why the namespace change in this PR? The description doesn't say anything

Reply:
There was a bug and it was failing while I was developing it. The e2e namespace may not exist when running must-gather (it can fail sooner).

Reply:
Update the PR description as well